### PR TITLE
Add a wrapper to wrap all optimization for mobile.

### PR DESCRIPTION
**Pull request by kimishpatel**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35227 Add a wrapper to wrap all optimization for mobile.**
* #35226 Refactor code to move const prop to convolution 2d replacer.

Summary:
This wraps.
1. Conv BN folding (not mobile specific)
2. insert XNNPACK conv2d/Linear ops
3. Remove prepacking ops.

Test Plan:
Checked via model transformation of one of the internal model.

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20603562](https://our.internmc.facebook.com/intern/diff/D20603562)

### CODE REVIEW

**Comment by dreiss** [comment]
Is there a strong reason to write this wrapper in C++?  Putting it in Python would make it easier to experiment with.

**Comment by kimishpatel** [comment]
> Is there a strong reason to write this wrapper in C++? Putting it in Python would make it easier to experiment with.

So each individual step of the wrapper pass is also available as a python binding. BatchNormConv2d fusion, freezing, insertPrePackedOp, FoldPrePackedOps.

This is just sugar around it.

**Comment by dreiss** [comment]
My vote, then, would be to just do the wrapper in Python as well, to make further work on it easier.

**Comment by kimishpatel** [comment]
> My vote, then, would be to just do the wrapper in Python as well, to make further work on it easier.

My preference is to keep in the `_jit_pass_*` space, since it is a JIT pass, wrapper around JIT passes to be precise, and should bleong there. If you want to keep it in python, I need to find an appropriate place for it (maybe `torch.nn.utils..`?), but in what ways, keeping it in python makes further work on it easier (not having to recompile)?

**Review by dreiss — APPROVED** [review]
